static/skin/viewer.js

```diff
@@ -262,10 +263,8 @@ function onClickEvent(e) {
   const target = matchingAncestorElement(e.target, iframeDocument, "a");
   if (target !== null && "href" in target) {
     if ( isExternalUrl(target.href) ) {
-      target.setAttribute("target", "_top");
-      if ( viewerSettings.linkBlockingEnabled ) {
-        return blockLink(target);
-      }
+      target.setAttribute("href", blockLink(target.href));
+      contentIframe.contentWindow.parent.postMessage({ externalURL : target.href }, "*");
```

Comment:
I don't understand the point of this. You are instructing the iframe window to send a message to the event handler below (`handleMessage`), which will simply give you a variable that you already have access to, i.e. `target.href`). Instead of using this circuitous route, why not just issue the action `window.location = target.href` directly here? What is the purpose of sending this message via the iframe, since nothing in the iframe needs to process it (and can't unless you inject a script into the iframe's document to deal with it)?

Sorry if I've misunderstood, as I haven't studied the full viewer code.

Reply:
> why not just issue the action `window.location = target.href` directly here?

This is what the previous code did and it was blocked because of sandboxing (since the action was performed in the context of the iframe which didn't have privileges of navigating the top level context). Now that action is initiated via a message - iframe sends a message and the top level receives and handles it in a different context where navigation is perfectly legal.

Reply:
OK. I understand that you can't navigate the iframe's contentWindow, but you don't need to that since you have access right here to `window`, which is the top-level window that is outside the iframe and is not sandboxed. It should be possible to navigate that window freely.

Reply:
No. Sandboxing applied to the iframe's document prevents it (and any script running in its name, which was the case here) from (directly) navigating the top-level window. Hence the need for indirection via `postMessage()`.

Reply:
@veloman-yunkan What you say would be true for a script running inside the iframe. However, `viewer.js` isn't running in the iframe, so it has access to and can navigate the top window, or open new windows/tabs. I did a quick check on this, and confirm it's possible to navigate the top window from `viewer.js`. I'll post the check in a separate comment, so as not to pollute the code view here.